### PR TITLE
resolved isues coming on POS Doctype and set value on new checkbox on…

### PR DESCRIPTION
**Pull request by Manjot-Vienna**
… order prepare it

### CODE REVIEW

File: ModelLibrary/Model/MOrder.cs [hunk 1]
```diff
@@ -3414,6 +3441,34 @@ public String CompleteIt()
                 MDocType dt = MDocType.Get(GetCtx(), GetC_DocType_ID());
                 DocSubTypeSO = dt.GetDocSubTypeSO();
 
+                //to check document type if it is pos then we need to check the document type on selected document types
+                if (MDocType.DOCSUBTYPESO_POSOrder.Equals(DocSubTypeSO))
+                {
+                    string paymentbaseType = Util.GetValueOfString(DB.ExecuteScalar("select va009_paymentbasetype from va009_paymentmethod where VA009_PaymentMethod_ID=" + GetVA009_PaymentMethod_ID(), null, Get_Trx()));
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MOrder.cs:3447]: Use upper Case for Keywords and standard notation for query variables

File: ModelLibrary/Model/MOrder.cs [hunk 2]
```diff
@@ -3414,6 +3441,34 @@ public String CompleteIt()
...
+                    if (!paymentbaseType.Equals(PAYMENTRULEPO_Check) && !paymentbaseType.Equals(PAYMENTRULEPO_LetterOfCredit) && !paymentbaseType.Equals(PAYMENTRULEPO_Cash) && !paymentbaseType.Equals(PAYMENTRULEPO_CashPlusCard))
+                    {
+                        if ((Get_ColumnIndex("C_DocTypePayment_ID") > 0) && (Get_ColumnIndex("C_BankAccount_ID") > 0))
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MOrder.cs:3450]: get these values from document type

File: ModelLibrary/Model/MOrder.cs (hunk 1 above)
Comment (Manjot-Vienna) [reply ModelLibrary/Model/MOrder.cs]: Changes Done

File: ModelLibrary/Model/MOrder.cs (hunk 2 above)
Comment (Manjot-Vienna) [reply ModelLibrary/Model/MOrder.cs]: Changes Done

**Review by Bharat-Vienna — APPROVED** [review]
OK

**Pull request merged**